lib/mongoid/scroll/base64_encoded_cursor.rb

```diff
@@ -0,0 +1,29 @@
+require 'base64'
+require 'json'
+
+module Mongoid
+  module Scroll
+    # Allows to serializer/deserialize the cursor using RFC 4648
+    class Base64EncodedCursor < Cursor
+      def to_s
+        Base64.strict_encode64({ value: super, field_type: field_type, field_name: field_name, direction: direction }.to_json)
+      end
+
+      class << self
+        def parse(str)
```

Comment:
In `Cursor` [this is an instance method](https://github.com/mongoid/mongoid-scroll/blob/master/lib/mongoid/scroll/cursor.rb#LL10C21-L10C21), and a cursor can be initialize from a value. Override that method.

```
class Base64EncodedCursor < Cursor
  def parse(value)
    # base64 implementation
  end
end
```

With this `Base64EncodedCursor` will work like `Cursor`, aka `Base64EncodedCursor.new(value, options ...)`.

Reply:
I tried something in 93185a37ac24780a45e2f7eb31f8b056f76687c8.

Basically what is done in `initialize` in `Cursor` doesn't make sense for `Base64EncodedCursor`. With a `Base64EncodedCursor`, what is set int the `options` hash is useless because everything is contained in the Base64 string. So every attribute that is set there will be overridden in `Base64EncodedCursor#parse`. I also had to update `Cursor.extract_field_options` to not raise when it's called for a `Base64EncodedCursor` and the `field` options are missing (since it's not required anymore).

Another way could be to just override `initialize` entirely in `Base64EncodedCursor` and do what is currently done in `Base64EncodedCursor#parse`. But it might be detrimental if we think that later on we'll have common parts in the initializers between all cursors.

Reply:
The third option is to pass options down, which avoids the base class change, I added a comment below.